guides/v2.1/coding-standards/technical-guidelines/technical-guidelines.md

```diff
@@ -471,6 +471,39 @@ We are reviewing this section and will publish it soon.
 
 13.4. Exception in a single CLI command SHOULD NOT break the CLI framework; running other commands SHOULD still be possible.
 
+## 14. Events
+
+14.1. All values (including objects) passed to event MUST NOT be modified in the event observer. Use plugins instead for this application.
+
+{% collapsible Examples: %}
+{%highlight php startinline=1%}
+
+```
+class SampleEventObserverThatModifiesInputs
+{
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        /** @var \Magento\Framework\App\DataObject $transport */
+        $transport = $observer->getData('transport');
+        
+        if ($transport->getData('some_value') === true) {
+            /**
+             * Expecting this value to go back to the original event dispatcher violates
+             * this rule. Other observers could change the data, or Magento could make
+             * architectural changes always sending immutable objects.
+             */
+            $transport->setData('output_return_value', true);
+        }
+    }
+}
+```
+{%endhighlight%}
+{% endcollapsible %}
+
+14.2. Use as specific events as possible. Don't use a `global` subscription to an event when the area impacted is just `frontend`.
```

Comment:
>Use as specific events as possible

May I suggest keeping the tone and style of the entire document: _Events used SHOULD be as specific as possible._

> Don't use a `global` subscription ... `frontend`.

Same about tone & style here: _A `global` subscription to an event SHOULD NOT be used when the area impacted is just `frontend`._